build_patch/xbattbar/xbattbar-darwin.diff

```diff
@@ -0,0 +1,115 @@
+ 
+ obj/stamp:
+ 	mkdir obj
+@@ -32,13 +31,11 @@
+ 
+ 
+ install: $(TARGET) $(APM_CHECK)
+-	install -d -m 0755 $(DESTDIR)/usr/lib/$(PROJECT)
+-	install -d -m 0755 $(DESTDIR)/usr/bin
+-	install -d -m 0755 $(DESTDIR)/usr/share/man/man1
+-	install -m 0755 $(APM_CHECK) $(DESTDIR)/usr/lib/$(PROJECT)/
+-	install -m 0755 xbattbar-check-acpi $(DESTDIR)/usr/lib/$(PROJECT)/
+-	install -m 0755 xbattbar-check-sys  $(DESTDIR)/usr/lib/$(PROJECT)/
+-	install -m 0755 $(TARGET) $(DESTDIR)/usr/bin/
+-	install -m 0644 xbattbar.man $(DESTDIR)/usr/share/man/man1/$(PROJECT).1 
++	$(INSTALL) -d -m 0755 $(DESTDIR)/usr/lib/$(PROJECT)
++	$(INSTALL) -d -m 0755 $(DESTDIR)/usr/bin
++	$(INSTALL) -d -m 0755 $(DESTDIR)/usr/share/man/man1
++	$(INSTALL) -m 0755 $(APM_CHECK) $(DESTDIR)/usr/lib/$(PROJECT)/
++	$(INSTALL) -m 0755 $(TARGET) $(DESTDIR)/usr/bin/
++	$(INSTALL) -m 0644 xbattbar.man $(DESTDIR)/usr/share/man/man1/$(PROJECT).1
+ 
+-include $(wildcard obj/*.d) 
++include $(wildcard obj/*.d)
+diff -urN a/xbattbar-check-darwin.m b/xbattbar-check-darwin.m
+--- a/xbattbar-check-darwin.m	1970-01-01 08:00:00.000000000 +0800
++++ b/xbattbar-check-darwin.m	2021-06-02 22:02:50.000000000 +0800
+@@ -0,0 +1,37 @@
++#import <Foundation/Foundation.h>
++#import <stdio.h>
++
++@interface BCBatteryDeviceController : NSObject
++@property (nonatomic,copy,readonly) NSArray * connectedDevices;
++
+++(id)sharedInstance;
++@end
++
++@interface BCBatteryDevice : NSObject
++@property (assign,getter=isCharging,nonatomic) BOOL charging;
++@property (assign,getter=isInternal,nonatomic) BOOL internal;
++@property (assign,nonatomic) long long percentCharge;
++@end
++
++int main() {
++        @autoreleasepool {
++                NSArray<BCBatteryDevice *> *batteries = [[BCBatteryDeviceController sharedInstance] connectedDevices];
++
++                BCBatteryDevice *internalBattery;
++
++                for (BCBatteryDevice *battery in batteries) {
++                        if (battery.internal) {
++                                internalBattery = battery;
++                                break;
++                        }
++                }
++
++                if (internalBattery) {
++                        printf("battery=%lld\nac_line=%s\n", internalBattery.percentCharge, internalBattery.charging ? "on" : "off");
++                        return EXIT_SUCCESS;
++                } else {
++                        fprintf(stderr, "No internal battery found.\n");
++                        return EXIT_FAILURE;
++                }
++        }
++}
+diff -urN a/xbattbar.c b/xbattbar.c
+--- a/xbattbar.c	2021-06-02 22:00:04.000000000 +0800
++++ b/xbattbar.c	2021-06-02 22:09:12.000000000 +0800
+@@ -75,7 +75,7 @@
+ char *OFFIN_C  = "blue";
+ char *OFFOUT_C = "red";
+ 
+-char *EXTERNAL_CHECK = "/usr/lib/xbattbar/xbattbar-check-apm";
++char *EXTERNAL_CHECK = "/usr/lib/xbattbar/xbattbar-check-darwin";
```

Comment:
Is that path actually hard coded?

Reply:
it is, i guess i should use SED instead with $(MEMO_PREFIX)$(MEMO_SUB_PREFIX)